addons/mail_plugin/models/res_partner_iap.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+
+from odoo import fields, models
+
+
+class ResPartnerIap(models.Model):
+    """Technical model which stores the response returned by IAP.
+
+    The goal of this model is to not enrich 2 times the same company. We do it in a
+    separate model to not add heavy field (iap_enrich_info) on the <res.partner>
+    model.
+
+    We also save the requested domain, so whatever the values are on the <res.partner>,
+    we will always retrieve the already enriched <res.partner> and the corresponding
+    IAP information.
+    """
+
+    _name = 'res.partner.iap'
+    _description = 'Partner IAP'
+
+    partner_id = fields.Many2one('res.partner', string='Partner', help='Corresponding partner',
+                                 ondelete='cascade', required=True)
+    search_domain = fields.Char('Search Domain / Email', help='Domain used to find the company')
```

Comment:
Just to match naming, we could name it iap_search_domain so that grepping it returns all its use and definitions on both partner and partner_iap models.

Reply:
Yes, good idea :)